application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/pages/GCConfigurationPage.java

```diff
@@ -85,58 +104,254 @@ public IDisplayablePage createPage(IPageDefinition dpd, StreamModel items, IPage
-		gcConfig.addAggregate(JdkAggregators.USE_DYNAMIC_GC_THREADS);
-		gcConfig.addAggregate(JdkAggregators.GC_TIME_RATIO_MIN);
-		gcConfigSection.setClient(gcConfig.getControl());
-
-		Section heapConfigSection = CompositeToolkit.createSection(container, toolkit,
-				Messages.GCConfigurationPage_SECTION_HEAP_CONFIG);
-		ItemAggregateViewer heapConfig = new ItemAggregateViewer(heapConfigSection, toolkit);
-		heapConfig.addAggregate(JdkAggregators.HEAP_CONF_INITIAL_SIZE_MIN);
-		heapConfig.addAggregate(JdkAggregators.HEAP_CONF_MIN_SIZE);
-		heapConfig.addAggregate(JdkAggregators.HEAP_CONF_MAX_SIZE);
-		heapConfig.addAggregate(JdkAggregators.USE_COMPRESSED_OOPS);
-		heapConfig.addAggregate(JdkAggregators.COMPRESSED_OOPS_MODE);
-		heapConfig.addAggregate(JdkAggregators.HEAP_ADDRESS_SIZE_MIN);
-		heapConfig.addAggregate(JdkAggregators.HEAP_OBJECT_ALIGNMENT_MIN);
-		heapConfigSection.setClient(heapConfig.getControl());
-
-		Section ycConfigSection = CompositeToolkit.createSection(container, toolkit,
-				Messages.GCConfigurationPage_SECTION_YOUNG_CONFIG);
-		ItemAggregateViewer ycConfig = new ItemAggregateViewer(ycConfigSection, toolkit);
-		ycConfig.addAggregate(JdkAggregators.YOUNG_GENERATION_MIN_SIZE);
-		ycConfig.addAggregate(JdkAggregators.YOUNG_GENERATION_MAX_SIZE);
-		ycConfig.addAggregate(JdkAggregators.NEW_RATIO_MIN);
-		ycConfig.addAggregate(JdkAggregators.TENURING_THRESHOLD_INITIAL_MIN);
-		ycConfig.addAggregate(JdkAggregators.TENURING_THRESHOLD_MAX);
-		ycConfig.addAggregate(JdkAggregators.USES_TLABS);
-		ycConfig.addAggregate(JdkAggregators.TLAB_MIN_SIZE);
-		ycConfig.addAggregate(JdkAggregators.TLAB_REFILL_WASTE_LIMIT_MIN);
-		ycConfigSection.setClient(ycConfig.getControl());
-
-		gcConfig.setValues(getDataSource().getItems());
-		heapConfig.setValues(getDataSource().getItems());
-		ycConfig.setValues(getDataSource().getItems());
-
-		addResultActions(form);
-
-		return null;
+		return new GCInformationUi(parent, toolkit, pageContainer, state);
+	}
+
+	// Straw man flag filter, need to support CMS, Serial, ParallelGC, Epsilon
+	// Looking at the old collector name in order to inspect/load a single event.
+	private IItemFilter collectorFlags(String oldCollector) {
+		// THis may happen for JFR files without GC configuration events, like those of async-profiler
+		if (oldCollector == null) {
+			return ItemFilters.all();
+		}
+
+		// Flags like ParallelGCThreads, ConcGCThreads, Xmx, NewRatio are left as this information is contained in the gc configuration event
+		// Most flags from https://github.com/openjdk/jdk11u/blob/master/src/hotspot/share/gc/shared/gc_globals.hpp
+		// are not added at this time.
+		switch (oldCollector) {
+		// TODO Cannot discover EpsilonGC via the old collector name, because Epsilon reports the same as SerialGC
+//		case "Epsilon":
+//			// from https://github.com/openjdk/jdk11u/blob/6c31ac2acdc2b2efa63fe92de8368ab964d847e9/src/hotspot/share/gc/epsilon/epsilon_globals.hpp
```

Comment:
Dead code. Want to try some other means of discovery?

Reply:
Yes, I'd like to, as I am not satisfied with the current approach.

Reply:
@thegreystone Finally I got some time to rework the discovery.

I'd like your input however : 
1. I created a class `GCFlagFilters` however I wonder if it makes sens to move it to `org.openjdk.jmc.flightrecorder` module in the `org.openjdk.jmc.flightrecorder.jdk` ?
2. I am not sure how to write tests. I have been playing with `org.openjdk.jmc.flightrecorder.uitest` module, but it's a bit slow to use, and I was only able to run them via maven command line.

Reply:
1. I think it is fine where it is, but you can strike up a conversion in the JMC slack/the jmc-dev list, and if there is enough interest, moving it to core could be a separate PR.

2. There is a launcher in org.openjdk.jmc.flightrecorder.uitest that should allow you to run only the flight recorder uitests from within Eclipse.